stacks-signer/src/runloop.rs

```diff
@@ -272,10 +272,17 @@ impl RunLoop {
 
     /// Refresh the signer configuration by retrieving the necessary information from the stacks node
     /// Note: this will trigger DKG if required
-    fn refresh_signers(&mut self, current_reward_cycle: u64) -> Result<(), ClientError> {
+    fn refresh_signers(
+        &mut self,
+        current_reward_cycle: u64,
+        in_prepare_phase: bool,
+    ) -> Result<(), ClientError> {
         let next_reward_cycle = current_reward_cycle.saturating_add(1);
         self.refresh_signer_config(current_reward_cycle);
-        self.refresh_signer_config(next_reward_cycle);
+        // don't try to refresh the next reward cycle's signer state if there's no state for that cycle yet.
+        if in_prepare_phase {
+            self.refresh_signer_config(next_reward_cycle);
+        }
```

Comment:
I don't think its _wrong_ to try to refresh the next cycle's state before the prepare phase, but it definitely results in lots of unnecessary RPC calls and log warnings.